flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphPartitionReleaseTest.java

```diff
@@ -110,6 +111,68 @@ public void testStrategyNotifiedOfFinishedVerticesAndResultsRespected() throws E
 		});
 	}
 
+	@Test
+	public void testStrategyNotifiedOfUnFinishedVertices() throws Exception {
+		// setup a pipeline of 3 failover regions (f1 -> f2 -> f3), where
```

Comment:
What is `f3`? According to the description `sourceVertex` is in `f1`, `o1`, `o2`, `o3` are in `f2`. There are no more operators.

Reply:
the comment is outdated, in a previous version there was another operator at the end, but this wasn't actually necessary for the test.